src/charm.py

```diff
@@ -1160,15 +1160,16 @@ def server_cert_sans_dns(self) -> List[str]:
             assert isinstance(target, str), target  # for type checker
             return [target]
 
+        targets: List[str] = [target] if target else []
         # This is an IP address. Try to look up the hostname.
         with contextlib.suppress(OSError, TypeError):
             name, _, _ = socket.gethostbyaddr(target)  # type: ignore
             # Do not return "hostname" like '10-43-8-149.kubernetes.default.svc.cluster.local'
             if is_hostname(name) and not name.endswith(".svc.cluster.local"):
-                return [name]
+                targets.append(cast(str, name))
 
         # If all else fails, we'd rather use the bare IP
-        return [target] if target else []
+        return targets
```

Comment:
Could you add a couple of test cases? I'm thinking:

- When external_hostname is set, then the CSR should be for the external hostname.
- When external_hostname is unset, then the CSR should be for the loadbalancer IP.